db/db.go

```diff
@@ -0,0 +1,260 @@
+// Copyright 2017, Square, Inc.
+
+// Package db is a connector to query a DB instance with a Query object passed
+// in.
+package db
+
+import (
+	"errors"
+
+	"github.com/square/etre/query"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+var (
+	// Unable to connect to DB
+	ErrConnect = errors.New("cannot connect to db")
+
+	// Error updating entity
+	ErrUpdateEntity = errors.New("error updating entity")
+
+	// Error deleting entity from DB
+	ErrDeleteEntity = errors.New("error deleting entity")
+
+	// Error not found
+	ErrNotFound = errors.New("not found")
+)
+
+// Map of Kubernetes Selection Operator
+// (https://github.com/kubernetes/apimachinery/blob/master/pkg/selection/operator.go)
+// to mongoDB Operator
+// (https://docs.mongodb.com/manual/reference/operator/query/#query-selectors).
+var operatorMap = map[string]string{
+	"in":    "$in",
+	"notin": "$nin",
+	"=":     "$eq",
+	"==":    "$eq",
+	"!=":    "$ne",
+	"lt":    "$lt",
+	"gt":    "$gt",
+}
+
+// Entity is a representation of what we store in the database.
+type Entity map[string]interface{}
+
+// Connector interface has methods needed to do CRUD operations on entities.
+type Connector interface {
+
+	// Managing a single entity
+	// TODO: CreateEntity(Entity) string
+	// TODO: ReadEntity(string) Entity
+	// TODO: UpdateEntity(string) Entity
+	// TODO: DeleteEntity(string) error
+
+	// Managing labels for a single entity
+	// TODO: ReadEntityLabels(string)
+	// TODO: DeleteEntityLabel(string, string)
+
+	// Managing Multiple entities
+	CreateEntities([]Entity) []string
+	ReadEntities(query.Query) ([]Entity, error)
+	UpdateEntities(query.Query, Entity) ([]Entity, error)
+	DeleteEntities(query.Query) ([]Entity, error)
+
+	// Managing connection to DB
+	Connect() error
+	Disconnect()
+}
+
+// mongo struct stores all info needed to connect and query a mongo instance.
+type mongo struct {
+	url        string
+	database   string
+	collection string
+	session    *mgo.Session
+}
+
+// NewConnector creates an instance of Connector.
+func NewConnector(url string, database string, collection string) Connector {
+	return &mongo{
+		url:        url,
+		database:   database,
+		collection: collection,
+	}
+}
+
+// Connect creates a session to db. All queries to DB will copy this session to
+// make a request (https://godoc.org/gopkg.in/mgo.v2#session.Copy).
+func (m *mongo) Connect() error {
+	session, err := mgo.Dial(m.url)
+	if err != nil {
+		return ErrConnect
```

Comment:
That makes a lot more sense. In my testing I was getting frustrated that the human friendly messages were lost but wasn't sure how to fix. I've updated the code to do what you suggested. Thanks.